rcgen/src/csr.rs

```diff
@@ -19,6 +19,13 @@ pub struct PublicKey {
 	alg: &'static SignatureAlgorithm,
 }
 
+impl PublicKey {
+	/// The algorithm used to generate the public key and sign the CSR.
+	pub fn alg(&self) -> &SignatureAlgorithm {
```

Comment:
For a new fn where we have some flexibility I think `algorithm` would be a better name and will resolve the mismatch you called out in your unit test w.r.t the `assert_eq!`. Let's do that?

Reply:
Done 🙂